airbyte-workers/src/main/java/io/airbyte/workers/WorkerUtils.java

```diff
@@ -52,8 +54,14 @@ public static void gentleClose(final Process process, final long timeout, final
       return;
     }
 
+    if (new EnvConfigs().getWorkerEnvironment().equals(WorkerEnvironment.KUBERNETES)) {
+      LOGGER.debug("Gently closing process {}", process.info().commandLine().get());
+    }
+
     try {
-      process.waitFor(timeout, timeUnit);
+      if (process.isAlive()) {
```

Comment:
Why is this check here?

Reply:
Before this change, `waitFor` was always called, which is why we saw the destination ports being released.

With our new change, where `getReturnCode` is handling this, this call is only necessary if the process is alive. It also better encodes the contract where if a process has an exit value, it means all associated resources have been closed.